source/server/http_test_server_filter.h

```diff
@@ -2,20 +2,68 @@
 
 #include <string>
 
+#include "envoy/common/time.h"
 #include "envoy/server/filter_config.h"
 
+#include "external/envoy/source/common/common/lock_guard.h"
+#include "external/envoy/source/common/common/thread.h"
+
 #include "api/server/response_options.pb.h"
 
 namespace Nighthawk {
 namespace Server {
 
-// Basically this is left in as a placeholder for further configuration.
+/**
+ * Filter configuration container class for the test server extension.
+ * Instances of this class will be shared accross instances of HttpTestServerDecoderFilter.
+ */
 class HttpTestServerDecoderFilterConfig {
 public:
+  /**
+   * Constructs a new HttpTestServerDecoderFilterConfig instance.
+   *
+   * @param proto_config The proto configuration of the filter.
+   */
   HttpTestServerDecoderFilterConfig(nighthawk::server::ResponseOptions proto_config);
+
+  /**
+   * @return const nighthawk::server::ResponseOptions& read-only reference to the proto config
+   * object.
+   */
   const nighthawk::server::ResponseOptions& server_config() { return server_config_; }
 
+  /**
+   * Gets the number of elapsed nanoseconds since the last call (server wide).
+   * Safe to use concurrently.
+   *
+   * @param time_source Time source that will be used to obain an updated monotonic time sample.
+   * @return uint64_t 0 on the first call, else the number of elapsed nanoseconds since the last
+   * call.
+   */
+  static uint64_t getElapsedNanosSinceLastRequest(Envoy::TimeSource& time_source);
+
 private:
+  /**
+   *  Utility class for thread safe tracking of elapsed monotonic time.
+   */
+  class ThreadSafeMontonicTimeStopwatch {
```

Comment:
Why is this class part of the HttpTestServerDecoderFilterConfig? is there anything the binds them together other than the fact that one uses the other? Or can we (for clarity and reusability) define it as a standalone one?

Reply:
I didn't want other things to rely on it yet, but if we feel this is good to go for that, then we can move it out. Maybe it should go into a separate file then and have direct test coverage. Let me know what you think.

Reply:
I think the idea of having it in its own file with test coverage is a good one and as a bonus we will end up with a simpler interface.